packages/babel-parser/test/fixtures/annex-b/enabled/assignmenttargettype/input.js

```diff
@@ -0,0 +1,12 @@
+function async() {}
+
+async() = 1;
+async() += 1;
+async()++;
+++async();
+for (async() in [1]) { }
+for (async() of [1]) {}
+
+async() &&= 1;
+
+import("")++;
```

Comment:
Since these examples are invalid in both annex-B and non-annex-B mode, could you extract these cases to `invalid-assignment-target-type`? The current one can also be renamed to `valid-*`.